reference_docs/YAML_models/ietf_v2_document_attributes.yaml

```diff
@@ -0,0 +1,524 @@
+    description: "Include boilerplate from Section 10.4(d) of http://tools.ietf.org/html/rfc2026".
+
+  linkmailto:
+    tags: ["Processing instructions for xml2rfc"]
+    title: "Generate mailto"
+    description: "Generate mailto: URL, as appropriate".
+
+  linefile:
+    tags: ["Processing instructions for xml2rfc"]
+    title: "linefile"
+    description: "A string like &quot;35:file.xml&quot; or just &quot;35&quot; (file name then defaults to the containing file's real name or to the latest linefile specification that changed it) that will be used to override xml2rfc's reckoning of the current input position (right after this PI) for warning and error reporting purposes (line numbers are 1-based)".
+
+  notedraftinprogress:
+    tags: ["Processing instructions for xml2rfc"]
+    title: "Note draft in progress"
+    description: 'Generates "(work in progress)", as appropriate'.
+
+  private:
+    tags: ["Processing instructions for xml2rfc"]
+    title: "private"
+    description: "Produce a private memo rather than an RFC or Internet-Draft".
+
+  refparent:
+    tags: ["Processing instructions for xml2rfc"]
+    title: "Ref parent"
+    description: "Title of the top-level section containing all references".
+
+  rfcedstyle:
+    tags: ["Processing instructions for xml2rfc"]
+    title: "rfcedstyle"
+    description: |
+      "Attempt to closely follow finer details from the latest observable RFC-Editor style so as to minimize the probability of being sent back corrections after submission.
+
+      This directive is a kludge whose exact behavior is likely to change on a regular basis to match the current flavor of the month; presently, it will:
+
+      * capitalize the adjective &quot;`This`&quot; in automatically generated headings,
+      * use the variant &quot;`acknowledgement`&quot; spelling instead of Merriam Webster's main &quot;`acknowledgment`&quot; dictionary entry,
+      * use the &quot;`eMail`&quot; spelling instead of Knuth's more modern &quot;`email`&quot; spelling,
+      * only put one blank line instead of two before top sections,
+      * omit &quot;`Intellectual Property and Copyright Statements`&quot; and &quot;`Author's Address`&quot; from the table of content, and
+      * not limit the indentation to a maximum tag length in `<references>` sections.".
+
+  rfcprocack:
+    tags: ["Processing instructions for xml2rfc"]
+    title: "rfcprocack"
+    description: "If there already is an automatically generated Acknowledg(e)ment section, pluralize its title and add a short sentence acknowledging that xml2rfc was used in the document's production to process an input XML source file in RFC-2629 format".
+
+  slides:
+    tags: ["Processing instructions for xml2rfc"]
+    title: "Slides"
+    description: "When producing a html file, produce multiple files for a slide show".
+
+  sort-refs:
+    tags: ["Processing instructions for xml2rfc"]
+    title: "Sort references"
+    description: "(`sortrefs`) sort references".
```

Comment:
Description is very vague. Do we know according to which criteria the references will be sorted? Alphabetically? In order of appearance?
```suggestion
    description: " Sort references according to CRITERIA."
```

Reply:
This attribute is used by xml2rfc:

https://datatracker.ietf.org/doc/html/rfc7991
<img width="596" alt="Screenshot 2021-07-04 at 1 38 14 PM" src="https://user-images.githubusercontent.com/11865/124374343-21e51600-dccd-11eb-8021-469bfea53670.png">

https://www.rfc-editor.org/materials/intro_xml2rfc_75.pdf
<img width="1031" alt="Screenshot 2021-07-04 at 1 38 33 PM" src="https://user-images.githubusercontent.com/11865/124374350-2d384180-dccd-11eb-9b60-c2e181922626.png">

An IETF RFC reference section looks like this:
<img width="605" alt="Screenshot 2021-07-04 at 1 39 35 PM" src="https://user-images.githubusercontent.com/11865/124374370-548f0e80-dccd-11eb-85ab-e1daf01848cd.png">
...
<img width="615" alt="Screenshot 2021-07-04 at 1 39 52 PM" src="https://user-images.githubusercontent.com/11865/124374372-5c4eb300-dccd-11eb-87fc-d510a28043bd.png">

I believe it sorts by the "reference tag" instead of by "reference title".

Reply:
The point being, the sorting is done by the preexisting IETF tools, and is out of the hands of Metanorma: Metanorma just passes on the sort-refs preprocessing instruction. IETF do define it that vaguely, but Ronald is clearly right.

Reply:
Thank you!